service/src/main/kotlin/io/provenance/explorer/domain/entities/Transactions.kt

```diff
@@ -474,3 +478,66 @@ class TxNftJoinRecord(id: EntityID<Int>) : IntEntity(id) {
     var metadataId by TxNftJoinTable.metadataId
     var metadataUuid by TxNftJoinTable.metadataUuid
 }
+
+object TxEventsTable : IntIdTable(name = "tx_events") {
+    val blockHeight = integer("block_height")
+    val txHash = varchar("tx_hash", 64)
+    val txHashId = reference("tx_hash_id", TxCacheTable)
+    val txMessageId = text("tx_message_id")
+    val txMsgTypeId = varchar("tx_message_type_id", 128)
+    val eventType = varchar("event_type", 256)
+    val txMessageHash = text("tx_message_hash")
+}
+
+class TxEventRecord(id: EntityID<Int>) : IntEntity(id) {
+    companion object : IntEntityClass<TxEventRecord>(TxEventsTable) {
+
+        private fun findByTxHashAndMessageHash(txHashId: Int, messageHash: String) = transaction {
+            TxEventRecord.find { (TxEventsTable.txHashId eq txHashId) and (TxEventsTable.txMessageHash eq messageHash) }
```

Comment:
Should match on `tx_message_id` instead of `tx_message_hash` (int vs text). Should also match on `event_type` as thats the uniqueness, tx_hash_id, tx_message_id, and event_type